### ISSUE TITLE
Parameter format byte breaks generation

### ISSUE THREAD

**Issue by nmcb** [zalando/play-swagger#20]
The parameter `format: byte` for `type: string` breaks generation.

API definition:
```
---
swagger: '2.0'
x-api-first-package: controllers
info:
  version: 0.0.0
  title: Simple API Test
paths:
  /{bippy}:
    get:
      parameters:
        - in: path
          name: bippy
          type: string
          format: byte
          # path parameter required as a workaround for issue 5 
          required: true
      responses:
        200:
          description: bippy response
# model definition required as a workaround for issue 18          
definitions:
  bop:
    properties:
      name:
        type: string
```

Compilation:
```
[play-swagger-test] $ compile
[info] Updating {file:/Users/marco/Code/play-swagger-test/}root...
[info] Resolving jline#jline;2.12.1 ...
[info] Done updating.
[info] Compiling 17 Scala sources and 1 Java source to /Users/marco/Code/play-swagger-test/target/scala-2.11/classes...
[error] /Users/marco/Code/play-swagger-test/conf/bippy.yaml:0: No URL path binder found for type Byte. Try to implement an implicit PathBindable for this type.
[error] /Users/marco/Code/play-swagger-test/conf/bippy.yaml:0: not enough arguments for method fromPath: (implicit binder: play.api.mvc.PathBindable[Byte])play.core.routing.Param[Byte].
[error] Unspecified value parameter binder.
[error] /Users/marco/Code/play-swagger-test/conf/bippy.yaml:0: No URL path binder found for type Byte. Try to implement an implicit PathBindable for this type.
[error] /Users/marco/Code/play-swagger-test/conf/bippy.yaml:0: not enough arguments for method implicitly: (implicit e: play.api.mvc.PathBindable[Byte])play.api.mvc.PathBindable[Byte].
[error] Unspecified value parameter e.
[error] /Users/marco/Code/play-swagger-test/conf/bippy.yaml:0: No URL path binder found for type Byte. Try to implement an implicit PathBindable for this type.
[error] /Users/marco/Code/play-swagger-test/conf/bippy.yaml:0: not enough arguments for method implicitly: (implicit e: play.api.mvc.PathBindable[Byte])play.api.mvc.PathBindable[Byte].
[error] Unspecified value parameter e.
[error] 6 errors found
[error] (compile:compileIncremental) Compilation failed
[error] Total time: 4 s, completed Oct 13, 2015 7:19:49 PM
```

**Comment by slavaschmidt** [comment]
Following format are not supported:
format: binary, type: string	any sequence of octets
Following format is supported but incorrect:
format: byte, type:string	
Should be: 	base64 encoded characters
Handled as: Numeric byte type

**Comment by slavaschmidt** [comment]
One relevant discussion can be seen here:

https://github.com/OAI/OpenAPI-Specification/issues/50

**Comment by slavaschmidt** [comment]
For now, it seems like type binary is only supposed too be supported as body parameter.
We could use mime type ```application/octet-stream``` as a marker that body must be encoded or decoded as a binary type.
For ```byte```we probably need to define appropriate mime type to recognise it as a body parameter. For path, query and header parameters there should be no problems with encoding and decoding.

**Comment by slavaschmidt** [comment]
Code generation is not broken anymore. Support for ```byte```format is added and tested. Support for ```binary```format is added but not well tested. The question is: do we need a special mime type to distinguish it or do we use something else?

**Comment by slavaschmidt** [comment]
Released in 0.1.6
